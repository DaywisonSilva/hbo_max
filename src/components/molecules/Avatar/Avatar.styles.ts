import { css } from "@stitches/core";

export const avatarContainer = css({
  width: "152px",
  height: "227px",
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  alignItems: "center",
});

export const avatarIcon = css({
  width: "90%",
  height: "60%",
  borderRadius: "50%",
});
export const avatarLegend = css({
  fontFamily: "sans-serif",
  fontWeight: 700,
});
