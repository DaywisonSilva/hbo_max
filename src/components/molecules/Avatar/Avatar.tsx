import React from "react";

import { avatarContainer, avatarIcon, avatarLabel } from "./Avatar.styles";

type AvatarProps = {
  src: string;
  alt: string;
  label: string;
};

export default function Avatar({ src, alt, label }: AvatarProps) {
  return (
    <div className={avatarContainer({})}>
      {/* eslint-disable-next-line @next/next/no-img-element */}
      <img
        className={avatarIcon({})}
        alt={`Esta imagem Ã© do filme: ${alt}`}
        src={`${src}`}
      />
      <p className={avatarLabel({})}>{label} </p>
    </div>
  );
}
