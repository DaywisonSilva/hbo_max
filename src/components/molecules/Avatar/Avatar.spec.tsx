import { render, screen } from '@testing-library/react'
import Avatar from './Avatar'
describe('<Avatar/>', () => {
  it('shound render correctly', () => {
    const { container } = render(<Avatar src='./image.jpg' alt='test' />)

    expect(container.firstChild).not.toBeNull()
  })

  it('shound render correctly label', () => {
    render(<Avatar src='./image.jpg' alt='test' label='test-label' />)

    const label = screen.getByTestId('label')

    expect(label.textContent?.trim()).toBe('test-label')
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Avatar src='./image.jpg' alt='test' label='test-label' />
    )

    expect(container).toMatchSnapshot()
  })
})
