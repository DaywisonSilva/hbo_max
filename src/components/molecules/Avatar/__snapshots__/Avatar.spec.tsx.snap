// Vitest Snapshot v1

exports[`<Avatar/> > should match snapshot 1`] = `
<div>
  <div
    class="c-gKZQys"
  >
    <img
      alt="Esta imagem é do filme: test"
      class="c-kRpKXR"
      src="./image.jpg"
    />
    <p
      class="c-iVfnJB"
      data-testid="label"
    >
      test-label
       
    </p>
  </div>
</div>
`;
